### DISCUSSION TITLE
Live training

### DISCUSSION THREAD

**Discussion by aaijazi** [nickcoutsos/keymap-editor#145] [Ideas]
Hi,

Thanks for building this awesome piece of software! I was hoping to extend it to support the ability to see your keypresses and have it live update, so that you can use it to help familiarize yourself with a new keyboard. 

Something like: https://www.zsa.io/live-training/ 

I'm curious if you've ever tried doing that yourself or if there's any pitfalls I should be aware of before I start working on it? Thanks!

**Comment by nickcoutsos** [comment]
Doing that for keycodes seems fairly trivial but the obvious distinction comes from handling key presses that trigger other behaviors like layer changes (or even handling input that could come from different keys) where lower level event data is needed from the keyboard firmware beyond just keyboard HID stuff.

That and especially the live update would require firmware support I couldn't even begin to attempt. If that's something you want to get involved in you should have a look at the `#studio-development` and `#studio-ideas` channels for ZMK's Discord community.
